WeakAuras/SubRegionTypes/Tick.lua

```diff
@@ -0,0 +1,232 @@
+  tick_color = {
+    display = L["Color"],
+    setter = "SetTickColor",
+    type = "color",
+  },
+  tick_placement_mode = {
+    display = L["Placement Mode"],
+    setter = "SetTickPlacementMode",
+    type = "list",
+    values = WeakAuras.tick_placement_modes,
+  },
+  tick_placement = {
+    display = L["Placement"],
+    setter = "SetTickPlacement",
+    type = "number",
+    validate = WeakAuras.ValidateNumeric,
+  },
+  automatic_length = {
+    display = L["Automatic Length"],
+    setter = "SetAutomaticLength",
+    type = "bool",
+    defaultProperty = true,
+  },
+  tick_thickness = {
+    display = L["Thickness"],
+    setter = "SetTickThickness",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 2,
+  },
+  tick_length = {
+    display = L["Length"],
+    setter = "SetTickLength",
+    type = "number",
+    min = 0,
+    bigStep = 1,
+    default = 30,
+  },
+}
+
+local auraBarTrueLeft = {
+  ["HORIZONTAL"] = "LEFT",
+  ["HORIZONTAL_INVERSE"] = "RIGHT",
+  ["VERTICAL"] = "TOP",
+  ["VERTICAL_INVERSE"] = "BOTTOM",
+}
+
+local auraBarTrueLeftInverse = {
+  ["HORIZONTAL"] = "RIGHT",
+  ["HORIZONTAL_INVERSE"] = "LEFT",
+  ["VERTICAL"] = "BOTTOM",
+  ["VERTICAL_INVERSE"] = "TOP",
+}
+
+local function create()
+  local subRegion = CreateFrame("FRAME", nil, UIParent)
+  subRegion.texture = subRegion:CreateTexture()
+  subRegion.texture:SetDrawLayer("ARTWORK", 3)
+  subRegion.texture:SetAllPoints(subRegion)
+  return subRegion
+end
+
+local function onAcquire(subRegion)
+  subRegion:Show()
+end
+
+local function onRelease(subRegion)
+  subRegion:Hide()
+end
+
+local funcs = {
+  ["Update"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+  end,
+  ["OrientationChanged"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+    self:SetTickLength(self.tick_length, true)
+    self:SetTickThickness(self.tick_thickness, true)
+  end,
+  ["OnSizeChanged"] = function(self)
+    self:SetTickPlacement(self.tick_placement, true)
+    self:SetTickLength(self.tick_length, true)
+  end,
+  SetVisible = function(self, visible)
+    if self.tick_visible ~= visible then
+      self.tick_visible = visible
+      if visible then
+        self:Show()
+      else
+        self:Hide()
+      end
+    end
+  end,
+  SetTickColor = function(self, r, g, b, a)
+    self.tick_color[1], self.tick_color[2], self.tick_color[3], self.tick_color[4] = r, g, b, a or 1
+    self.texture:SetColorTexture(r, g, b, a or 1)
+  end,
+  SetTickPlacementMode = function(self, placement_mode)
+    if self.placement_mode ~= placement_mode then
+      self.placement_mode = placement_mode
+      self:SetTickPlacement(self.tick_placement)
+    end
+  end,
+  SetTickPlacement = function(self, placement, forced)
+    if forced or self.tick_placement ~= placement then
+      self.tick_placement = placement
+      self.parentOrientation = self.parent.effectiveOrientation
+      self.parentInverse = self.parent.inverseDirection
```

Comment:
Unfortunately inverseDirection is more complex than this:
There are two "inverses" for progress bars, and they cancel out if both are set.
(Ignoring the third inverse via the orientation)

a) The inverse setting set in the options of the aura, this is inverseDirection
b) A inverse setting of the duration information, this is used by e.g. the Cast trigger to automatically inverse the direction of the progress bar for channels. (Whereas by default a cast fills up a cast bar, a channel depletes.)

There are two bugs here:
* This second inverse is currently not taking into account, you can easily figure out whether the duration is inversed by looking copying state.inverse in Update
* The SetInverse function needs to Notify you that the Inverse flag has changed

Also I would prefer it if you introduced simple getters in the progress bar for these properties:
self.parent:GetEffectiveOrientation + self.parent:GetInverseDirection()

(I often forget to do that, but it's cleaner to use functions instead of directly using the variables.)